### PR TITLE
feat(security): initial version of security

### PR DESCRIPTION
**Pull request by billxinli**
This is an initial implementation of the security work.

I still need to update the README, which I will when the code is solidified, still contemplating about breaking up the security library that parses the configurations. 

Quick primer 
```
const spec = require('./some-spec.json')
const API = require('./lib')(spec)
// define root server url
const api = new API('https://httpbin.org', {
  secret: 'some-secret',
  jwt: {
    exp: '5m'
  }
})
async function main () {
  // auto generated methods matching your OAS operationIds

  // Allows per method override if needed
  await api.someMethod({
    secret: 'some-secret',
    jwt: {
      exp: '5m'
    }
  })

  // Or just use default security configuration
  await api.someMethod()
}
main()
```

### CODE REVIEW

File: lib/security/index.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+               */
+              switch (securityScheme.bearerFormat) {
+                case 'JWT': {
+                  const { exp } = jwt || {}
+                  const jwtOptions = {}
+                  if (exp) {
+                    jwtOptions.expiresIn = exp
+                  }
+
+                  headers.authorization.push(`Bearer ${await signJWT({}, secret, jwtOptions)}`)
```
Comment (billxinli) [inline lib/security/index.js:59]: Right now we just sign on an blank object. Still need to discuss this. 

Sign blank object
Sign on body for non GET
Sign on query for GET

As the latter two sends more information than required.

File: lib/security/jwt.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const jwt = require('jsonwebtoken')
+
+async function signJWT (payload, secret, options) {
```
Comment (ahmadnassri) [inline lib/security/jwt.js:3]:
```js
module.exports = async function
```

no need for an object wrapper?

File: lib/security/index.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  }
+
+  return {
+    headers,
+    queries
+  }
+}
+
+module.exports = { parseSecurity }
```
Comment (ahmadnassri) [inline lib/security/index.js:110]: why do we need to wrap in an object? just export the function directly ...

**Review by ahmadnassri — CHANGES_REQUESTED** [review]

File: lib/security/index.js (hunk 1 above)
Comment (ahmadnassri) [reply lib/security/index.js:59]: what if the `jwt` param in options indicates a `payload` property that can have either: `query`, `body`, or a custom object as needed?

**Pull request closed without merging**